generators/csharp/codegen/src/context/AbstractCsharpGeneratorContext.ts

```diff
@@ -296,6 +296,20 @@ export abstract class AbstractCsharpGeneratorContext<
         }
     }
 
+    public isPrimitive(typeReference: TypeReference): boolean {
+        switch (typeReference.type) {
+            case "container":
+                return typeReference.container.type === "optional";
+            case "named": {
+                return false;
```

Comment:
should we be handling aliases to primitives here?

Reply:
Yes, great call - I should have noticed that in the `isOptional` helper above.